inference/models/gaze/gaze.py

```diff
@@ -0,0 +1,334 @@
+        """
+        x_px = max(min(math.floor(normalized_x * image_width), image_width - 1), 0)
+        y_px = max(min(math.floor(normalized_y * image_height), image_height - 1), 0)
+        return x_px, y_px
+
+    def _make_response(self, faces: List[Detection], gazes: List[Tuple[float, float]], imgW: int, imgH: int,
+                       time_load_img: float, time_face_det: float,
+                       time_gaze_det: float) -> GazeDetectionInferenceResponse:
+        """Prepare response object from detected faces and corresponding gazes.
+
+        Args:
+            faces (List[Detection]): The detected faces.
+            gazes (List[tuple(float, float)]): The detected gazes.
+            imgW (int): The width (px) of original image.
+            imgH (int): The height (px) of original image.
+            time_load_img (float): The processing time for loading image.
+            time_face_det (float): The processing time for face detection.
+            time_gaze_det (float): The processing time for gaze detection.
+
+        Returns:
+            GazeDetectionInferenceResponse: The response object including the detected faces and gazes info.
+        """
+        predictions = []
+        for i in range(len(faces)):
+            face = faces[i]
+            gaze = gazes[i]
+
+            landmarks = []
+            for keypoint in face.keypoints:
+                x, y = self._normalized_to_pixel_coordinates(keypoint.x, keypoint.y, imgW, imgH)
+                landmarks.append(Point(x=x, y=y))
+
+            bbox = face.bounding_box
+            x_center = bbox.origin_x + bbox.width / 2
+            y_center = bbox.origin_y + bbox.height / 2
+            score = face.categories[0].score
+
+            prediction = GazeDetectionPrediction(
+                face=FaceDetectionPrediction(x=x_center, y=y_center, width=bbox.width, height=bbox.height,
+                                             confidence=score, class_name="face", landmarks=landmarks),
+                pitch=gaze[0], yaw=gaze[1])
+            predictions.append(prediction)
+
+        response = GazeDetectionInferenceResponse(predictions=predictions,
+                                                  time_total=(time_load_img + time_face_det + time_gaze_det),
+                                                  time_load_img=time_load_img, time_face_det=time_face_det,
+                                                  time_gaze_det=time_gaze_det)
+        return response
+
+    def _detect_gaze(self, pil_imgs: List[Image.Image]) -> List[Tuple[float, float]]:
+        """Detect faces and gazes in an image.
+
+        Args:
+            pil_imgs (List[Image.Image]): The PIL image list, each image is a cropped facial image.
+
+        Returns:
+            List[Tuple[float, float]]: Pitch (degree) and Yaw (degree).
+        """
+        ret = []
+        for i in range(0, len(pil_imgs), GAZE_MAX_BATCH_SIZE):
```

Comment:
Nice! Great way of respecting the GAZE_MAX_BATCH_SIZE without dropping any predictions.